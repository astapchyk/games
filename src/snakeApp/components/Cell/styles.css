.cell {
  position: absolute;
  width: calc(100%/16);
  height: calc(100%/16);
}

.food {
  background-color: yellow;
}

.snake {
  background-color: lavender;
  z-index: 1;
}
