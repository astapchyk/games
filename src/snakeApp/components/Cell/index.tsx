import React, { ReactElement, memo } from 'react';

import styles from './styles.css';
export { styles };

import { getClassNames } from '../../utils/common';

type CellProps = {
  cell: Cell;
  className: string;
};

type Cell = {
  x: number;
  y: number;
};

// eslint-disable-next-line react/display-name
const Cell = memo(
  ({ cell, className }: CellProps): ReactElement => {
    return (
      <div
        style={{
          left: `${cell.x * 5}vh`,
          top: `${cell.y * 5}vh`,
        }}
        className={getClassNames(styles.cell, className)}
      />
    );
  },
);

export default Cell;
