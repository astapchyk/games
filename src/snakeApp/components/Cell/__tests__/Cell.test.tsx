import React from 'react';
import { render } from '@testing-library/react';

import Cell from '../index';

describe('<Cell />', () => {
  test('renders the component', async () => {
    const cell = { x: 1, y: 2 };
    const cellName = 'food';

    const { container } = render(<Cell cell={cell} className={cellName} />);
    expect(container).toMatchSnapshot();
  });
});
