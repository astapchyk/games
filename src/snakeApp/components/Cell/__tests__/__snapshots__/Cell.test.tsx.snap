// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> renders the component 1`] = `
<div>
  <div
    class="cell food"
    style="left: 5vh; top: 10vh;"
  />
</div>
`;
